assets/javascript/app.js

```diff
@@ -196,10 +188,78 @@
 
   var template = document.getElementById('template').innerHTML
 
-
-
   function applyData(data){
     document.getElementById('content').innerHTML = Handlebars.compile(template)(data);
   }
 
   applyData(data);
+
+function handleSaveNoteClick() {
+  var body = getNoteDetails()
+
+  fetch('/save-note', {
+    method: 'POST',
+    body: JSON.stringify(body),
+    headers: new Headers({
+      'Content-Type': 'application/json'
+    })
+  }).then(function(response) {
+    return response.json()
+  }).then(function(data) {
+    applyData(data)
+  }).then(function () {
+    reapplyImageZoom(body.path.caption, body.userJourney.title)
+    registerHandlers()
+  })
+}
+
+function reapplyImageZoom(caption, title) {
+  var activeJourneyElement = Array.from(document.querySelectorAll('.image-set-title')).find(function (el) {
+    // remove any leading numbers that match the format (1. ) incl. space
+    return el.innerText.replace(/\d+\.\s/g, "") === title
+  })
+
+  var activeImageElement = Array.from(activeJourneyElement.parentElement.querySelectorAll('.image')).find(function (el) {
```

Comment:
😷
we may want to consider rewriting this project with a framework if this app gets much bigger